daft/runners/ray_runner.py

```diff
@@ -536,8 +547,25 @@ def place_in_queue(item):
                             elif len(next_step.instructions) == 0:
                                 logger.debug("Running task synchronously in main thread: %s", next_step)
                                 assert isinstance(next_step, SingleOutputPartitionTask)
+                                [single_partial] = next_step.partial_metadatas
```

Comment:
This seems new - why was this necessary when before we didn't need to ensure that `num_rows` is available?